iotileship/iotile/ship/resources/filesystem_manager.py

```diff
@@ -0,0 +1,120 @@
+''' A Resource for operating on a Filesystem
+
+    Can work on either a directory or a unmounted block device
+
+    Important: mounting and unmounting of a block device requires the udisksctl utility, which is included with
+    most distributions of Ubuntu
+
+    All Steps using this Resource will use self.root as the root directory to find other files
+
+    Root directory is a pathlib.Path
+
+'''
+
+import subprocess
+import pathlib
+import sys
+
+from iotile.core.utilities.schema_verify import DictionaryVerifier
+from iotile.core.utilities.schema_verify import StringVerifier
+from iotile.core.exceptions import ArgumentError
+from .shared_resource import SharedResource
+
+RESOURCE_ARG_SCHEMA = DictionaryVerifier(desc="filesystem_manager arguments")
+RESOURCE_ARG_SCHEMA.add_required("res_path", StringVerifier("path to resource to work with. "
+                                                            "Specify file:// for a directory or block:// for an unmounted disk partition"))
+
+
+class FilesystemManagerResource(SharedResource):
+    '''
+    Arguments:
+        path (str): Required. One of:
+            file://<path> : Path to a directory that will be worked on
+            block://<path> : Path to a block device to work on (typically /dev/sd...)
+    '''
+
+
+    ARG_SCHEMA = RESOURCE_ARG_SCHEMA
+    _allowed_types = ["file", "block"]
+
+    def __init__(self, args):
+        super(FilesystemManagerResource, self).__init__()
+
+        self._path = args.get('res_path')
+
+        if not "://" in self._path:
+            raise ArgumentError("Cannot detect resource type (looking for something like 'file://)'",
+                                parameter_name="res_path")
+
+        self._type = self._path.split('://')[0]
+        self._resource_loc = self._path.split('://')[1]
+
+        if self._type not in self._allowed_types:
+            raise ArgumentError("FilesystemManagerResource unknown filesystem type", parameter_name="res_path")
+
+        if self._type == "block" and sys.platform != "linux":
+            raise ArgumentError("block;// paths are only supported on Linux. \
+                Mount the device and pass the filepath", parameter_name="res_path")
+
+        if self._type == "block" and sys.platform == "linux":
+            try:
+                subprocess.run(['udisksctl', 'help'], stdout=subprocess.DEVNULL, check=True)
+            except subprocess.CalledProcessError:
+                print("Could not find 'udisksctl'. Make sure it's installed")
+                raise
+
+        self.root = None
+        self._unmount_in_close = False
+
+    #If self._resource_loc is mounted, returns pathlib.Path to mount location
```

Comment:
There should usually be a space between the `#` and the text of a comment. https://realpython.com/python-pep8/#block-comments

(this comment applies throughout the PR)